core/src/main/java/io/grpc/NameResolver.java

```diff
@@ -99,14 +99,21 @@ public void refresh() {}
     public static final Attributes.Key<Integer> PARAMS_DEFAULT_PORT =
         Attributes.Key.create("params-default-port");
 
+    /**
+     * Proxy detector used in name resolution.
+     */
+    @ExperimentalApi("https://github.com/grpc/grpc-java/issues/5113")
```

Comment:
@zhangkun83 should this be a TransportAttr?

Reply:
Waiting for your feedback on this